internal/server/smb/server.go

```diff
@@ -5,36 +5,63 @@ import (
 	"fmt"
 
 	"github.com/kubernetes-csi/csi-proxy/client/apiversion"
+	fsserver "github.com/kubernetes-csi/csi-proxy/internal/server/filesystem"
 	"github.com/kubernetes-csi/csi-proxy/internal/server/smb/internal"
+	"k8s.io/klog"
 )
 
 type Server struct {
-	hostAPI API
+	hostAPI  API
+	fsServer *fsserver.Server
 }
 
 type API interface {
-	NewSmbGlobalMapping(remotePath, username, password string) error
+	IsSmbMapped(remotePath string) (bool, error)
+	SMBLink(remotePath, localPath string) error
+	NewSmbGlobalMapping(remotePath, localPath, username, password string) error
```

Comment:
the localPath in here and the localPath in above SMBLink is different, right?

Reply:
They are not different. Having said that we don't need the local path in the host APIs since its really not used any more since the link part is moved out. Will update the PR.